Add new entries for Makefile and update db test suite

albertoleal opened the pull request:
Hi @flavioribeiro!

This PR includes the following commits:

- Add new entries to Makefile for building a docker image, run tests inside a container;
- Use "shared example" to test the database suite for the current implementations: in memory and mongo, without repeating the suite.

Cheers

flavioribeiro commented:
Awesome, thanks for that @albertoleal! Can you please fix the build  and add the new instructions to README before I merge? 🍻

albertoleal commented:
Fixed! 

RE: the new instructions on Makefile, it turned out that it wasn't working properly (building/executing tests in a docker container) and I removed them.

flavioribeiro commented:
Thank you! Cheers!
